common/lockmanager_test.go

```diff
@@ -0,0 +1,355 @@
+    // until key1 gets unlocked for the first time.
+    go func() {
+        defer threadFinishBlock.Done();
+
+        threadLockBlock.Done();
+
+        Lock(key1);
+        err := Unlock(key1);
+        if (err != nil) {
+            test.Fatalf("Failed to unlock for the second time.");
+        }
+    }()
+
+    // Wait for the second thread to try to lock key1
+    // while the lock is being used by the first thread.
+    threadLockBlock.Wait();
+    
+    // Small sleep to ensure the thread tries to lock key1.
+    time.Sleep(10 * time.Millisecond);
+
+    // Unlock for the first time.
+    err := Unlock(key1);
+    if (err != nil) {
+        test.Fatalf("Failed to unlock for the first time.");
+    }
+    
+    // Wait for the thread to lock and unlock key1.
+    threadFinishBlock.Wait();
+}
+
+func TestStaleLockWithNonStaleLock(test *testing.T) {
+    Clear();
+    defer Clear();
+
+    key1 := "testkey1";
+    staleDuration := time.Duration(config.STALELOCK_DURATION_SECS.Get()) * time.Minute;
+    var threadExecutionBlock sync.WaitGroup;
+    var timestampBlock sync.WaitGroup;
+
+    Lock(key1);
+    Unlock(key1);
+
+    val, _ := lockMap.Load(key1); 
+    // Set the timestamp for the lock to be considered stale.
+    val.(*lockData).timestamp = time.Now().Add(-1 * (staleDuration + (1 * time.Minute)));
+
+    threadExecutionBlock.Add(1);
+    timestampBlock.Add(1);
+
+    go func() {
+        defer threadExecutionBlock.Done();
+
+        // Wait for the lock's timestamp to be reset.
+        timestampBlock.Wait();
+
+        RemoveStaleLocksOnce();
+    }()
+
+    // Reset the locks timestamp.
+    Lock(key1);
+    Unlock(key1);
+    
+    // Let the goroutine continue its execution after resetting the timestamp.
+    timestampBlock.Done();
+
+    // Wait for the go routine to finish its execution.
+    threadExecutionBlock.Wait();
+    
+    // Check if the lock still exists in the lock map.
+    _, exists := lockMap.Load(key1);
+    // Test if the non stale lock got removed.
+    if (!exists) {
+        test.Fatalf("Failed to retain lock even though it was accessed concurrently.");
+    }
+} 
+
+func TestStaleLockWithLockedKey(test *testing.T) {
+    Clear();
+    defer Clear();
+
+    key1 := "testkey1";
+    staleDuration := time.Duration(config.STALELOCK_DURATION_SECS.Get()) * time.Second;
+    var staleCheckBlock sync.WaitGroup;
+
+    Lock(key1);
+
+    val, _ := lockMap.Load(key1); 
+    // Set the timestamp for a lock to be considered stale.
+    val.(*lockData).timestamp = time.Now().Add(-1 * (staleDuration + (1 * time.Second)));
+    
+    staleCheckBlock.Add(1);
+
+    // Start a goroutine to check if the lock is stale.
+    go func() {
+        defer staleCheckBlock.Done();
+
+        RemoveStaleLocksOnce();
+    }()
+    
+    // Wait for the goroutine to finish its execution.
+    staleCheckBlock.Wait();
+
+    // Check if the lock still exists in the lock map and return it.
```

Comment:
I don't like this comment.
It talks about returning, but we are not returning anything
(the function we calls returns something, but we don't usually talk like that).

I also don't like how the comment above the conditional breaks the flow of code a bit.

To fix both of these, merge this comment with the one two lines down and put them both on this line.
This way, we have a higher quality comment and an uninterrupted block of code.

Note that this happens in a few places (at least the second part), look for them in this file.